shared/teams/main/team-row.tsx

```diff
@@ -0,0 +1,159 @@
+    <>
+      <Kb.ListItem2
+        type="Small"
+        firstItem={firstItem}
+        onClick={onViewTeam}
+        icon={<Kb.Avatar size={32} teamname={teamMeta.teamname} isTeam={true} />}
+        height={Styles.isMobile ? 90 : undefined}
+        body={
+          <Kb.Box2 direction="horizontal" fullHeight={true} fullWidth={true} style={styles.bodyContainer}>
+            <Kb.Box2 direction="horizontal" fullHeight={true} alignItems="center" style={styles.bodyLeft}>
+              <Kb.Box2
+                direction="vertical"
+                fullHeight={true}
+                alignItems="flex-start"
+                style={styles.bodyLeftText}
+              >
+                <Kb.Box2 direction="horizontal" gap="xtiny" alignSelf="flex-start" alignItems="center">
+                  <Kb.Text type="BodySemibold" lineClamp={1} ellipsizeMode="middle">
+                    {teamMeta.teamname}
+                  </Kb.Text>
+                  {teamMeta.isOpen && (
+                    <Kb.Meta
+                      title="open"
+                      backgroundColor={Styles.globalColors.green}
+                      style={styles.openMeta}
+                    />
+                  )}
+                </Kb.Box2>
+                <Kb.Text type="BodySmall">
+                  {teamMeta.memberCount.toLocaleString()} {pluralize('member', teamMeta.memberCount)}
+                </Kb.Text>
+                {Styles.isMobile && activity}
+              </Kb.Box2>
+            </Kb.Box2>
+            {!Styles.isMobile && (
+              <Kb.Box2 direction="horizontal" fullHeight={true} style={styles.bodyRight}>
+                {activity}
+              </Kb.Box2>
+            )}
+          </Kb.Box2>
+        }
+        action={
+          <Kb.Box2 direction="horizontal" gap={Styles.isMobile ? 'tiny' : 'xtiny'}>
+            {showChat && (
+              <Kb.Button
+                type="Dim"
+                onClick={onChat}
+                disabled={!teamMeta.isMember}
+                mode="Secondary"
+                small={true}
+                icon="iconfont-chat"
+                tooltip={!teamMeta.isMember ? 'You are not a member of this team.' : ''}
+              />
+            )}
+            <Kb.Button
+              type="Dim"
+              onClick={() => setShowingPopup(true)}
+              mode="Secondary"
+              small={true}
+              icon="iconfont-ellipsis"
+              tooltip=""
+              ref={popupRoot}
+            />
+          </Kb.Box2>
+        }
+        onlyShowActionOnHover="fade"
+      />
+      {popup}
+    </>
+  )
+}
+
+type ActivityLevel = 'active' | 'recently' | 'extinct'
+const activityToIcon: {[key in ActivityLevel]: Kb.IconType} = {
+  active: 'iconfont-fire',
+  extinct: 'iconfont-rip',
+  recently: 'iconfont-team-leave',
+}
+const activityToLabel = {
+  active: 'Active',
+  extinct: 'Extinct',
+  recently: 'Recently active',
+}
+const Activity = ({level}: {level: ActivityLevel}) => (
```

Comment:
can we factor this out into a common place? I made something like this in the teams nav header too.